src/upload-lib.ts

```diff
@@ -16,6 +16,11 @@ import { parseRepositoryNwo, RepositoryNwo } from "./repository";
 import * as util from "./util";
 import { SarifFile, ConfigurationError, wrapError } from "./util";
 
+const GENERIC_403_MSG =
+  "The repo on which this action is running is not opted-in to CodeQL code scanning.";
+const GENERIC_404_MSG =
+  "Not authorized to use the CodeQL code scanning feature on this repo.";
```

Comment:
I haven't verified this, but are you sure that you have the correct 404 and 403 error messages? 403 errors usually mean _forbidden_, but you have it here as 404.

Also, _not authorized_ (401) typically means that the user isn't logged in. I think it is better to use the phrase _forbidden_, which correlates better with 403.

```suggestion
const GENERIC_404_MSG =
  "The CodeQL code scanning feature is forbidden on this repository.";
```

Reply:
The strings were merely moved. I make no claim to their accuracy.